File: rust/src/bin/9/9.palindrome-number.rs
/*
 * @lc app=leetcode id=9 lang=rust
 *
 * [9] Palindrome Number
 */

// @lc code=start
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        if x.to_string().chars().rev().eq(x.to_string().chars()) {
            true
        } else {
            false
        }
    }
}
// @lc code=end

File: typescript/9/9.palindrome-number.ts
/*
 * @lc app=leetcode id=9 lang=typescript
 *
 * [9] Palindrome Number
 */

// @lc code=start
function isPalindrome(x: number): boolean {
    const arr = x.toString().split("");

    if (arr.toString() === arr.reverse().toString()) {
        return true;
    }
    return false;
};
// @lc code=end
